src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -688,6 +688,11 @@ function ( $all_circular, $circular_pair ) {
 
 		$dependents_to_deactivate = array_unique( $dependents_to_deactivate );
 
+		// Bail early if there are no dependents to deactivate.
+		if ( empty( $dependents_to_deactivate ) ) {
+			return;
+		}
+
```

Comment:
This could go above the `array_unique()` call since that will only be necessary if the array isn't empty (well, if it's greater than one element).

Reply:
I guess it's true that `array_unique` with a non-empty array could not result in an empty array. Addressed in 750b219.

Reply:
Is the native PHP call to `array_unique` really that costly?

Reply:
Probably negligible most of the time, but if we know that we can avoid calling it here I don't see any reason not to.